File: app/src/main/java/fr/insapp/insapp/IntroActivity.java
package fr.insapp.insapp;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;

import com.github.paolorotolo.appintro.AppIntro2;

import fr.insapp.insapp.fragments_intro.IntroClubsFragment;
import fr.insapp.insapp.fragments_intro.IntroEventsFragment;
import fr.insapp.insapp.fragments_intro.IntroNewsFragment;
import fr.insapp.insapp.fragments_intro.IntroNotificationsFragment;
import fr.insapp.insapp.fragments_intro.IntroProfileFragment;

/**
 * Created by thoma on 02/12/2016.
 */
public class IntroActivity extends AppIntro2 {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        addSlide(new IntroNewsFragment());
        addSlide(new IntroEventsFragment());
        addSlide(new IntroClubsFragment());
        addSlide(new IntroNotificationsFragment());
        addSlide(new IntroProfileFragment());

        showSkipButton(false);
        setProgressButtonEnabled(true);
        setVibrate(false);
    }

    @Override
    public void onSkipPressed(Fragment currentFragment) {
        super.onSkipPressed(currentFragment);

        Intent i = new Intent(this, LegalConditionsActivity.class);
        i.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        startActivity(i);

        finish();
    }

    @Override
    public void onDonePressed(Fragment currentFragment) {
        super.onDonePressed(currentFragment);

        Intent i = new Intent(this, LegalConditionsActivity.class);
        i.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        startActivity(i);

        finish();
    }
}

File: app/src/main/java/fr/insapp/insapp/activities/IntroActivity.kt
package fr.insapp.insapp.activities

import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.github.paolorotolo.appintro.AppIntro2
import fr.insapp.insapp.fragments.intro.*

/**
 * Created by thomas on 02/12/2016.
 */
class IntroActivity : AppIntro2() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        addSlide(IntroPostsFragment())
        addSlide(IntroEventsFragment())
        addSlide(IntroAssociationsFragment())
        addSlide(IntroNotificationsFragment())
        addSlide(IntroProfileFragment())

        showSkipButton(false)
        isProgressButtonEnabled = true
        setVibrate(false)
    }

    override fun onSkipPressed(currentFragment: Fragment?) {
        super.onSkipPressed(currentFragment)

        val i = Intent(this, LegalConditionsActivity::class.java)
        i.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
        startActivity(i)

        finish()
    }

    override fun onDonePressed(currentFragment: Fragment?) {
        super.onDonePressed(currentFragment)

        val i = Intent(this, SignInActivity::class.java)
        i.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
        startActivity(i)

        finish()
    }
}
